### PR TITLE
Swapped out another layer of requests from the first request and move…

### PR DESCRIPTION
**Pull request by PanopticaRising**
…d to a second request using includes.

### CODE REVIEW

**Comment by PanopticaRising** [comment]
Hide whitespace changes.

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -3717,18 +3703,46 @@ You should be able to reply to the student's email address (${options.student.pr
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         const data: any = mainData?.get({plain: true});
 
-        await mainData?.questions?.asyncForEach(async (question, i) => 
+        await mainData?.questions?.asyncForEach(async (question, i) =>
             await question.grades?.asyncForEach(async (grade, j) => {
-                if (_.isNil(grade.lastInfluencingAttempt) || _.isNil(grade.lastInfluencingAttempt.studentGradeInstance)) {
-                    logger.debug('No lastInfluencingAttempt and no studentGradeInstance exists');
+                if (_.isNil(grade.lastInfluencingCreditedAttemptId)) {
+                    console.log(data.questions[i].grades[j]);
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:3709]: no console logs

File: src/features/courses/course-controller.ts [hunk 2]
```diff
@@ -3717,18 +3703,46 @@ You should be able to reply to the student's email address (${options.student.pr
...
+                    logger.error('No lastInfluencingCreditedAttemptId. Cannot find the best version.');
                     return;
                 }
-                const probAttach = await grade.lastInfluencingAttempt.studentGradeInstance.getProblemAttachments();
-                grade.lastInfluencingAttempt.studentGradeInstance.problemAttachments = probAttach;
-                data.questions[i].grades[j].lastInfluencingAttempt.studentGradeInstance.problemAttachments = probAttach.map(pA => pA.get({plain: true}));
+
+                const gradeInstanceAttachments = await StudentWorkbook.findOne({
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:3714]: Why not us a mixin 🤷‍♂️

**Review by red-tt-bear — APPROVED** [review]

File: src/features/courses/course-controller.ts (hunk 2 above)
Comment (PanopticaRising) [reply src/features/courses/course-controller.ts:3713]: I gave the mixin a try, and it didn't pull anything back. I updated the PR with the mixin function.

**Pull request merged**